README.md

```diff
@@ -49,6 +48,62 @@ def example_s3_upload_flow():
 example_s3_upload_flow()
 ```
 
+### Write and run a flow with AwsCredentials and S3Bucket
+
+```python
+import asyncio
+from prefect import flow
+from prefect_aws AwsCredentials
+from prefect_aws.s3 import S3Bucket
+
+@flow
+async def aws_s3_bucket_roundtrip(creds):
+
+    s3_bucket = S3Bucket(
+        bucket_name="bucket",  # must exist
+        aws_credentials=creds,
+        basepath="subfolder",
+    )
+
+    key = await s3_bucket.write_path("data.csv", content=b"hello")
+
+    return await s3_bucket.read_path(key)
+
+# create an AwsCredentials block here or through UI
+aws_creds = AwsCredentials(
+    aws_access_key_id=AWS_ACCESS_KEY_ID,
+    aws_secret_access_key=AWS_SECRET_ACCESS_KEY
+)
+
+asyncio.run(aws_s3_bucket_roundtrip(aws_creds))
+```
```

Comment:
I'm confused by this. The credentials are created before the function aws_s3_bucket_roundtrip() is run, no?
```
# create an AwsCredentials block here or through UI
aws_creds = AwsCredentials(
    aws_access_key_id=AWS_ACCESS_KEY_ID,
    aws_secret_access_key=AWS_SECRET_ACCESS_KEY
)

# use the credentials you created to run flow
asyncio.run(aws_s3_bucket_roundtrip(aws_creds))
```

Reply:
The idea being that you may very well already have an AwsCredentials block stored, and you can reap the benefits of that by not needing to instantiate it in your flow as you already have it.

Reply:
Having a block as an input to a flow makes it a flow parameter which would make creating a deployment for this example flow difficult. I believe that you would need to enter all the values for credentials for each deployment run or have default values for your credentials. I think either would expose your credentials. I haven't tested this out, but it would be worth checking before using this example as is. If you want to be able to use different credentials chosen at invocation, I would recommend passing in a block document name and loading the block as part of the flow.

Reply:
I think once aws_creds is nested within the task, `creds` as a required argument should be deleted. Also, AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY should be strings

```

import asyncio
from prefect import flow
from prefect_aws AwsCredentials
from prefect_aws.s3 import S3Bucket
@flow
async def aws_s3_bucket_roundtrip():
    # create an AwsCredentials block here or through UI
    aws_creds = AwsCredentials(
        aws_access_key_id="AWS_ACCESS_KEY_ID",
        aws_secret_access_key="AWS_SECRET_ACCESS_KEY"
    )
    s3_bucket = S3Bucket(
        bucket_name="bucket",  # must exist
        aws_credentials=aws_creds,
        basepath="subfolder",
    )
    key = await s3_bucket.write_path("data.csv", content=b"hello")
    return await s3_bucket.read_path(key)

asyncio.run(aws_s3_bucket_roundtrip())
```

Reply:
I think we have the MinIO example where we pass a block document name, so this one will be a good example of instantiating a new block and using it.

Reply:
Update README.md

Reply:
I think you need to drop the "creds" as a requirement.

Reply:
Good thinking, Andrew

[Update README](https://github.com/PrefectHQ/prefect-aws/pull/47/commits/ad870d143b32287922c4313778e1f7fadfe87d5f)